src/Microsoft.AspNet.Server.Kestrel/Filter/FilteredStreamAdapter.cs

```diff
@@ -28,8 +28,13 @@ public class FilteredStreamAdapter
             _filteredStream = filteredStream;
             _socketInputStream = new SocketInputStream(SocketInput);
 
-            _filteredStream.CopyToAsync(_socketInputStream).ContinueWith((task, state) =>
+            var block = memory.Lease();
+            // Use pooled block for copy
+            _filteredStream.CopyToAsync(_socketInputStream, block).ContinueWith((task, state) =>
             {
+                var returnedBlock = task.Result;
+                returnedBlock.Pool?.Return(returnedBlock);
```

Comment:
In this case, since we aren't providing a minimum size to `memory.Lease()`, value it pool should never be null.